src/Calculator/Controls/CalculationResult.cpp

```diff
@@ -322,48 +329,24 @@ void CalculationResult::OnScrollClick(Object ^ sender, RoutedEventArgs ^ /*e*/)
     }
 }
 
-void CalculationResult::OnPointerEntered(Platform::Object ^ sender, PointerRoutedEventArgs ^ e)
-{
-    if (e->Pointer->PointerDeviceType == PointerDeviceType::Mouse && m_textBlock->ActualWidth >= m_textContainer->ActualWidth)
-    {
-        UpdateScrollButtons();
-    }
-}
-
-void CalculationResult::ShowHideScrollButtons(::Visibility vLeft, ::Visibility vRight)
-{
-    m_scrollLeft->Visibility = vLeft;
-    m_scrollRight->Visibility = vRight;
-}
-
 void CalculationResult::UpdateScrollButtons()
 {
-    // When the width is smaller than the container, don't show any
-    if (m_textBlock->ActualWidth < m_textContainer->ActualWidth)
+    if (m_textContainer == nullptr)
     {
-        ShowHideScrollButtons(::Visibility::Collapsed, ::Visibility::Collapsed);
-    }
-    // We have more number on both side. Show both arrows
-    else if (m_textContainer->HorizontalOffset > 0 && m_textContainer->HorizontalOffset < (m_textContainer->ExtentWidth - m_textContainer->ViewportWidth))
-    {
-        ShowHideScrollButtons(::Visibility::Visible, ::Visibility::Visible);
-    }
-    // Width is larger than the container and left most part of the number is shown. Should be able to scroll left.
-    else if (m_textContainer->HorizontalOffset == 0)
-    {
-        ShowHideScrollButtons(::Visibility::Collapsed, ::Visibility::Visible);
+        return;
     }
-    else // Width is larger than the container and right most part of the number is shown. Should be able to scroll left.
+
+    if (m_scrollLeft != nullptr)
     {
-        ShowHideScrollButtons(::Visibility::Visible, ::Visibility::Collapsed);
+        m_scrollLeft->Visibility = m_textContainer->HorizontalOffset > SCROLL_BUTTONS_APPROXIMATION_RANGE ? ::Visibility::Visible : ::Visibility::Collapsed;
```

Comment:
>SCROLL_BUTTONS_APPROXIMATION_RANGE  [](start = 71, length = 35)

It's probably worth taking your pr comment and putting it in here as an actual comment.  From the code itself, it isn't immediately obvious why this buffer is here.

Reply:
Comment added just before the definition of SCROLL_BUTTONS_APPROXIMATION_RANGE!